docs/FAQ.md

```diff
@@ -44,3 +44,37 @@ or client side to unmarshal a response) the default value is used to set the
 value of missing fields. Note that if the attribute is required then the
 generated code returns an error if the corresponding field is missing. So this
 only applies for non required attributes with default values.
+
+# How are views for a result type computed?
+
+Views can be defined on a result type. If a method returns a result type with
+multiple views then a views package is generated at the service level which
+projects the result type as per the attributes defined in the view.
+Constructors are generated in the service package to convert a result type
+to a viewed result type and vice versa.
+
+The views package defines a type with name as the result type name suffixed
```

Comment:
For each result type with multiple views the `views` package defines a type with identical field names and types but using pointers for all fields so that view specific validation logic may be generated.